src/common/options.cc

```diff
@@ -8640,6 +8640,40 @@ std::vector<Option> get_mds_client_options() {
         "If no filesystem name is given in mount() or this setting, the default "
         "filesystem will be mounted (usually the first created)."),
 
+    // MSGR On-wire compression
+
+    Option("ms_osd_compress_mode", Option::TYPE_STR, Option::LEVEL_ADVANCED)
+    .set_default("force")
+    .set_enum_allowed({"none", "force"})
+    .add_service("osd")
+    .set_flag(Option::FLAG_RUNTIME)
+    .set_description("Compression policy to use in MSGR for communicating with OSD")
+    .add_see_also("ms_compress_secure"),
+
+    Option("ms_osd_compress_min_size", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
+    .set_default(4_K)
+    .add_service("osd")
+    .set_flag(Option::FLAG_RUNTIME)
+    .set_description("Minimal message size eligable for on-wire compression")
+    .add_see_also("ms_osd_compress_mode"),
+
+    Option("ms_osd_compression_algorithm", Option::TYPE_STR, Option::LEVEL_ADVANCED)
+    .set_default("snappy zlib zstd lz4")
```

Comment:
does this imply that we can use snappy and other compression algorithm in the same time? if yes, which one is used under which circumstances?

Reply:
Only one algorithm can be chosen per connection. This list represents algorithms prioritization. I'll add additional information as a long description